File: app/src/main/java/com/example/mealrecipeapp/ui/profile/ProfileViewModel.java
package com.example.mealrecipeapp.ui.profile;

import androidx.lifecycle.ViewModel;

import com.example.mealrecipeapp.data.repository.AppRepository;

public class ProfileViewModel extends ViewModel {
    private final AppRepository appRepository;

    public ProfileViewModel(AppRepository appRepository) {
        this.appRepository = appRepository;
    }

    public String getUserName() {
        return appRepository.getUserName();
    }

    public String getUserEmail() {
        return appRepository.getUserEmail();
    }

    public String getUserImage() {
        return appRepository.getUserImage();
    }

    public void signOut() {
        appRepository.deleteUser();
    }
}

File: app/src/main/java/com/example/mealrecipeapp/ui/profile/ProfileViewModel.kt
package com.example.mealrecipeapp.ui.profile

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.mealrecipeapp.data.repository.AppRepository
import kotlinx.coroutines.launch

class ProfileViewModel(private val appRepository: AppRepository) : ViewModel() {
    fun getUserName(): String = appRepository.getUserName()
    fun getUserEmail(): String = appRepository.getUserEmail()
    fun getUserImage(): String = appRepository.getUserImage()
    fun getCheckRootSetting(): Boolean = appRepository.getCheckRootSetting()

    fun setCheckRootSetting(isChecked: Boolean) {
        appRepository.setCheckRootSetting(isChecked)
    }

    fun getCheckEmulatorSetting(): Boolean = appRepository.getCheckEmulatorSetting()

    fun setCheckEmulatorSetting(isChecked: Boolean) {
        appRepository.setCheckEmulatorSetting(isChecked)
    }

    fun getCheckUSBDebugSetting(): Boolean = appRepository.getCheckUSBDebugSetting()

    fun setCheckUSBDebugSetting(isChecked: Boolean) {
        appRepository.setCheckUSBDebugSetting(isChecked)
    }

    fun getCheckAccessibilitySetting(): Boolean = appRepository.getCheckAccessibilitySetting()

    fun setCheckAccessibilitySetting(isChecked: Boolean) {
        appRepository.setCheckAccessibilitySetting(isChecked)
    }

    fun signOut() {
        viewModelScope.launch {
            appRepository.deleteUser()
        }
    }
}
